src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/SparkSessionTests.cs

```diff
@@ -65,5 +68,95 @@ public void TestSignaturesV2_4_X()
+                {
+                    new StructField("Name", new StringType()),
+                    new StructField("Age", new IntegerType())
+                });
+                DataFrame df = _spark.CreateDataFrame(data, schema);
+                ValidateDataFrame(data.Select(a => a.Values), df, schema);
+            }
+
+            // Calling CreateDataFrame(IEnumerable<string> _) without schema
+            {
+                var data = new List<string>(new string[] { "Alice", "Bob" });
+                var schema = new StructType(new List<StructField>()
+                {
+                    new StructField("_1", new StringType())
+                });
+
+                DataFrame df = _spark.CreateDataFrame(data);                
+                ValidateDataFrame(data.Select(a => new object[] { a }), df, schema);
+            }
+
+            // Calling CreateDataFrame(IEnumerable<int> _) without schema
+            {
+                var data = new List<int>(new int[] { 1, 2 });
+                var schema = new StructType(new List<StructField>()
+                {
+                    new StructField("_1", new IntegerType())
+                });
+
+                DataFrame df = _spark.CreateDataFrame(data);
+                ValidateDataFrame(data.Select(a => new object[] { a }), df, schema);
+            }
+
+            // Calling CreateDataFrame(IEnumerable<double> _) without schema
+            {
+                var data = new List<double>(new double[] { 1.2, 2.3 });
+                var schema = new StructType(new List<StructField>()
+                {
+                    new StructField("_1", new DoubleType())
+                });
+
+                DataFrame df = _spark.CreateDataFrame(data);
+                ValidateDataFrame(data.Select(a => new object[] { a }), df, schema);
+            }
+
+            // Calling CreateDataFrame(IEnumerable<bool> _) without schema
+            {
+                var data = new List<bool>(new bool[] { true, false });
+                var schema = new StructType(new List<StructField>()
+                {
+                    new StructField("_1", new BooleanType())
+                });
+
+                DataFrame df = _spark.CreateDataFrame(data);
+                ValidateDataFrame(data.Select(a => new object[] { a }), df, schema);
+            }
+        }
+
+        /// <summary>
+        /// Validates that the dataframe returned by the CreateDataFrame API is correct as per the
+        /// given data.
+        /// </summary>
+        /// <param name="data">Given data for the dataframe</param>
+        /// <param name="df">Dataframe Object to validate</param>
+        /// <param name="schema">Expected schema of the dataframe</param>
+        private void ValidateDataFrame(IEnumerable<object[]> data, DataFrame df, StructType schema)
+        {
+            Assert.Equal(schema, df.Schema());
+            Row[] rows = df.Collect().ToArray();
```

Comment:
Can we do something like:
```C#
Assert.True(df.Collect().Select(r => r.Values).SequenceEqual(data))
```
?

Reply:
This is throwing an error of failure
Message: 
    Assert.True() Failure
    Expected: True
    Actual:   False

Reply:
do you know why? it's a bit weird to mix `foreach` and counter.

Reply:
Does the following also give you an error?

```csharp
Assert.Equal(data, df.Collect().Select(r => r.Values));
```

Reply:
^ this worked